static/app/components/globalDrawer/components.tsx

```diff
@@ -0,0 +1,110 @@
+import {useCallback, useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import type {DrawerOptions} from 'sentry/components/globalDrawer/types';
+import SlideOverPanel from 'sentry/components/slideOverPanel';
+import {IconClose} from 'sentry/icons/iconClose';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+
+export interface DrawerPanelProps extends Required<Omit<DrawerOptions, 'onOpen'>> {
+  children: React.ReactNode;
+  isOpen?: boolean;
+  onOpen?: () => void;
+}
+
+export function DrawerPanel({
+  children,
+  isOpen = false,
+  onClose,
+  onOpen,
+  closeOnOutsideClick = true,
+  closeOnEscapeKeypress = true,
+}: DrawerPanelProps) {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(isOpen);
+  const panelRef = useRef<HTMLDivElement>(null);
+
+  useOnClickOutside(panelRef, () => {
+    if (isOpen && closeOnOutsideClick) {
+      onClose();
+      setIsDrawerOpen(false);
+    }
+  });
```

Comment:
The handler function probably should be in a `useCallback` otherwise this effect is going to re-run every render (https://github.com/getsentry/sentry/blob/6e3cddfd060aa6db961af54197718331cc7a531b/static/app/utils/useOnClickOutside.tsx#L37)